Client/core/CSettings.cpp

```diff
@@ -3541,6 +3591,16 @@ void CSettings::SaveData()
     CVARS_GET("browser_remote_websites", bOldRemoteWebsites);
     CVARS_GET("browser_remote_javascript", bOldRemoteJavascript);
 
+    // Main Menu settings
+    CVARS_SET("mainmenu_hide_news", m_pHideNews->GetSelected());
+    CVARS_SET("mainmenu_hide_quickconnect", m_pHideQuickConnect->GetSelected());
+    CVARS_SET("mainmenu_hide_hostgame", m_pHideHostGame->GetSelected());
+    CVARS_SET("mainmenu_hide_mapeditor", m_pHideMapEditor->GetSelected());
+    CVARS_SET("mainmenu_auto_open_browser", m_pAutoOpenServerBrowser->GetSelected());
+
+    CMainMenu* pMainMenu = CLocalGUI::GetSingleton().GetMainMenu();
+    pMainMenu->ReloadMenuItems();
```

Comment:
This place is crashing the game for some reason :P, any idea?

Reply:
You need to do `m_menuItems.clear()` after the for loop in `CMainMenu:DeleteMenuItems`. It deletes the CEGUI instances but doesn't clear the item queue.